lib/internal/repl.js

```diff
@@ -127,7 +127,8 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     }
 
     if (data) {
-      repl.history = data.split(/[\n\r]+/, repl.historySize);
+      repl.history = data.trim().split(/[\n\r]+/)
+        .reverse().slice(0, repl.historySize);
```

Comment:
Umm… I think the intent of `repl.historySize` is to limit the file size, not the length of the history that’s being used?